evm/src/cpu/kernel/asm/hash/blake2/blake2_f.asm

```diff
@@ -55,14 +55,22 @@ global blake2_f:
     %rep 8
         SWAP1
         DUP2
-        %mstore_current_general
+        %swap_mstore
         %increment
     %endrep
     // stack: start + 8, rounds, t0, t1, flag, retdest
 
+    // Retrieve the offset in `start + 8`
```

Comment:
Any reason why you retrieve the offset instead of keeping it in address form and replacing the `%mstore_current_general` l.82 with a `%swap_mstore`?

Reply:
Not sure why actually, I changed it as you suggested!